packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -510,18 +560,25 @@ class _AppBarState extends State<AppBar> {
 
     final double toolbarHeight = widget.toolbarHeight ?? kToolbarHeight;
 
+    final Color backgroundColor = widget.backgroundColor
```

Comment:
Shouldn't these color calculations use the `brightness` property of the appBar itself if it has one?

Reply:
That's a good question. The existing AppBar/AppBarTheme brightness property is (and was) [only used to set the SystemUiOverlayStyle](https://github.com/flutter/flutter/pull/69251/files#diff-20c19d254b0a82027625f1246ba51cb2264bfabf105fe91d532c19e6a6d77478R768).  This is not correctly documented, I'll fix that.

In this case the brightness of theme's color scheme is used because we're choosing default colors that contrast with the theme's colors, which is also as things were.